src/NJsonSchema/Generation/DefaultReflectionService.cs

```diff
@@ -178,137 +174,150 @@ public JsonTypeDescription GetDescription(Type type, IEnumerable<Attribute> pare
         }
 
         /// <summary>Checks whether a type is nullable.</summary>
-        /// <param name="type">The type.</param>
-        /// <param name="parentAttributes">The parent attributes (e.g. property or parameter attributes).</param>
+        /// <param name="contextualType">The type.</param>
         /// <param name="defaultReferenceTypeNullHandling">The default reference type null handling used when no nullability information is available.</param>
         /// <returns>true if the type can be null.</returns>
-        public virtual bool IsNullable(Type type, IEnumerable<Attribute> parentAttributes, ReferenceTypeNullHandling defaultReferenceTypeNullHandling)
+        public virtual bool IsNullable(ContextualType contextualType, ReferenceTypeNullHandling defaultReferenceTypeNullHandling)
         {
-            var jsonPropertyAttribute = parentAttributes?.OfType<JsonPropertyAttribute>().SingleOrDefault();
+            var jsonPropertyAttribute = contextualType.GetContextAttribute<JsonPropertyAttribute>();
             if (jsonPropertyAttribute != null && jsonPropertyAttribute.Required == Required.DisallowNull)
+            {
                 return false;
+            }
 
-            if (parentAttributes.TryGetIfAssignableTo("NotNullAttribute", TypeNameStyle.Name) != null)
+            if (contextualType.ContextAttributes.TryGetAssignableToTypeName("NotNullAttribute", TypeNameStyle.Name) != null)
+            {
                 return false;
+            }
 
-            if (parentAttributes.TryGetIfAssignableTo("CanBeNullAttribute", TypeNameStyle.Name) != null)
+            if (contextualType.ContextAttributes.TryGetAssignableToTypeName("CanBeNullAttribute", TypeNameStyle.Name) != null)
+            {
                 return true;
+            }
 
-            if (type.Name == "Nullable`1")
-                return true;
+            if (contextualType.Nullability != Nullability.Unknown)
```

Comment:
Add and only run on defaultReferenceTypeNullHandling.Default also update Nswag with that (response)